src/runmode-unix-socket.c

```diff
@@ -355,6 +379,111 @@ static TmEcode UnixSocketPcapFilesCheck(void *data)
     TmThreadContinueThreads();
     return TM_ECODE_OK;
 }
+
+/**
+ * \brief Command to add a file to treatment list
+ *
+ * \param cmd the content of command Arguments as a json_t object
+ * \param answer the json_t object that has to be used to answer
+ * \param data pointer to data defining the context here a PcapCommand::
+ */
+TmEcode UnixSocketPcapFolderAdd(json_t *cmd, json_t* answer, void *data)
+{
+    PcapCommand *this = (PcapCommand *) data;
+    const char *folder;
+#ifdef OS_WIN32
+    struct _stat st;
+#else
+    struct stat st;
+#endif /* OS_WIN32 */
+
+    json_t *jarg = json_object_get(cmd, "folder");
+    if(!json_is_string(jarg)) {
+        SCLogInfo("error: command is not a string");
+        json_object_set_new(answer, "message", json_string("command is not a string"));
+        return TM_ECODE_FAILED;
+    }
+    folder = json_string_value(jarg);
+#ifdef OS_WIN32
+    if(_stat(folder, &st) != 0) {
+#else
+    if(stat(folder, &st) != 0) {
+#endif /* OS_WIN32 */
+        json_object_set_new(answer, "message", json_string("Folder does not exist"));
+        return TM_ECODE_FAILED;
+    }
+
+    if (this->currentfolder == NULL)
+    {
+        this->currentfolder = SCStrdup(folder);
+        return TM_ECODE_OK;
+    }
+    else
+    {
+        json_object_set_new(answer, "message", json_string("Previously set folder is still queued"));
+        return TM_ECODE_FAILED;
+    }
+}
+
+
+/**
+ * \brief Handle the file queue
+ *
+ * This function check if there is currently a file
+ * being parse. If it is not the case, it will start to
+ * work on a new file. This implies to start a new 'pcap-file'
+ * running mode after having set the file and the output dir.
+ * This function also handles the cleaning of the previous
+ * running mode.
+ *
+ * \param this a UnixCommand:: structure
+ * \retval 0 in case of error, 1 in case of success
+ */
+TmEcode UnixSocketPcapFolderCheck(void *data)
+{
+    PcapCommand *this = (PcapCommand *) data;
+    if (unix_manager_folder_task_running == 1) {
+        return TM_ECODE_OK;
+    }
+    if (unix_manager_folder_task_failed == 1) {
+        if (unix_manager_folder_task_failed) {
+            SCLogInfo("Preceeding task failed, cleaning the running mode");
+        }
+        unix_manager_folder_task_failed = 0;
+        if (this->currentfolder) {
+            SCFree(this->currentfolder);
+        }
+        this->currentfolder = NULL;
+
+        PostRunDeinit(RUNMODE_PCAP_FOLDER, NULL /* no ts */);
+        return TM_ECODE_OK;
+    }
+
+    if (this->currentfolder == NULL)
```

Comment:
style: { on same line for code blocks, on new line for function decl. See https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Coding_Style